dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/Filter/FilterFixtureBase.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Globalization;
+using Microsoft.Extensions.VectorData;
+using VectorDataSpecificationTests.Support;
+using Xunit;
+
+namespace VectorDataSpecificationTests.Filter;
+
+public abstract class FilterFixtureBase<TKey> : IAsyncLifetime
+    where TKey : notnull
+{
+    private int _nextKeyValue = 1;
+    private List<FilterRecord<TKey>>? _testData;
+
+    protected virtual string StoreName => "FilterTests";
+
+    protected abstract TestStore TestStore { get; }
+
+    protected virtual string DistanceFunction => Microsoft.Extensions.VectorData.DistanceFunction.CosineSimilarity;
+    protected virtual string IndexKind => Microsoft.Extensions.VectorData.IndexKind.Flat;
+
+    public virtual async Task InitializeAsync()
+    {
+        await this.TestStore.ReferenceCountingStartAsync();
+
+        this.Collection = this.TestStore.DefaultVectorStore.GetCollection<TKey, FilterRecord<TKey>>(this.StoreName, this.GetRecordDefinition());
+
+        if (await this.Collection.CollectionExistsAsync())
+        {
+            await this.Collection.DeleteCollectionAsync();
+        }
+
+        await this.Collection.CreateCollectionAsync();
+        await this.SeedAsync();
+
+        // Some databases upsert asynchronously, meaning that our seed data may not be visible immediately to tests.
```

Comment:
I am glad that you bring your DB experience, so I don't need to spend few days debugging why my tests are flaky ;)

Reply:
Ahahahaha well actually, most databases I'm used to (relational and to a lesser extent, document databases) don't typically have this problem - it's *usually* important for inserted rows to be immediately visible. Vector databases are a bit special, since querying and updating are very very asymmetrical: they're done by completely different sides of the production system, and typically you don't need immediate visibility of data. So this is a tiny bit new to me too...!